src/main/java/com/cloudbees/jenkins/plugins/file_leak_detector/FileHandleDump.java

```diff
@@ -94,11 +99,27 @@ public HttpResponse doActivate(@QueryParameter String opts) throws Exception {
 
         int exitCode = p.waitFor();
 
-        if (exitCode!=0)
-            // As the real cause of the failure of the agent is lost and we cannot access it, we can only recommend to
-            // see the logs.
-            throw new Error("Failed to activate file leak detector.\nPerhaps wrong parameters.\n" +
-                    "See logs for more info.\nSpecifically, look for 'Agent failed to start!' or something related\n\n");
+        if (exitCode!=0) {
+            /*
+            There are 2 high-level ways the process can fail:
+            1. The process can fail in org.kohsuke.file_leak_detector.Main#main. For example, this can happen if Jenkins
+               is running with a JRE instead of a full JDK and so the instrumentation API doesn't load. In this case,
+               the output of the process we created here contains the root cause of the failure. Because it is a
+               separate process, its output will _not_ be sent to Jenkins' logs/stdout/stderr by default, and so we
```

Comment:
I think we could use `Process#inheritIO` to make this happen automagically, but I am not familiar enough with that method to know about any quirks it might have.